core/src/lib/StitchError.ts

```diff
@@ -70,8 +72,10 @@ export default class StitchError {
    */
   private static handleRichError(response: Response, body: string): string {
     if (
-      response.headers[Headers.CONTENT_TYPE] !== undefined ||
-      response.headers[Headers.CONTENT_TYPE] !== ContentTypes.APPLICATION_JSON
+      response.headers[Headers.CONTENT_TYPE] === undefined ||
```

Comment:
Regarding my other comment, this is checking for the all lowercase version of the header (which is correct here), the headers from the response need to be preprocessed for the rich error to be properly decoded.